node/src/components/block_synchronizer/block_builder/tests.rs

```diff
@@ -0,0 +1,198 @@
+    let mut builder = BlockBuilder::new(
+        block.header().block_hash(),
+        false,
+        1,
+        TimeDiff::from_seconds(1),
+        TimeDiff::from_seconds(1),
+    );
+
+    let honest_peer = NodeId::random(&mut rng);
+    let dishonest_peer = NodeId::random(&mut rng);
+
+    assert!(builder
+        .handle_acceptance(None, Ok(Some(Acceptance::NeededIt)))
+        .is_ok());
+    assert!(builder.peer_list().qualified_peers(&mut rng).is_empty());
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder
+        .handle_acceptance(None, Ok(Some(Acceptance::HadIt)))
+        .is_ok());
+    assert!(builder.peer_list().qualified_peers(&mut rng).is_empty());
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder.handle_acceptance(None, Ok(None)).is_ok());
+    assert!(builder.peer_list().qualified_peers(&mut rng).is_empty());
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder
+        .handle_acceptance(Some(honest_peer), Ok(None))
+        .is_ok());
+    assert!(builder.peer_list().qualified_peers(&mut rng).is_empty());
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder
+        .handle_acceptance(Some(honest_peer), Ok(Some(Acceptance::HadIt)))
+        .is_ok());
+    assert!(builder.peer_list().qualified_peers(&mut rng).is_empty());
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder
+        .handle_acceptance(Some(honest_peer), Ok(Some(Acceptance::NeededIt)))
+        .is_ok());
+    assert!(builder
+        .peer_list()
+        .qualified_peers(&mut rng)
+        .contains(&honest_peer));
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder
+        .handle_acceptance(None, Err(BlockAcquisitionError::InvalidStateTransition))
+        .is_err());
+    assert!(builder
+        .peer_list()
+        .qualified_peers(&mut rng)
+        .contains(&honest_peer));
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder
+        .handle_acceptance(
+            Some(dishonest_peer),
+            Err(BlockAcquisitionError::InvalidStateTransition)
+        )
+        .is_err());
+    assert!(builder
+        .peer_list()
+        .qualified_peers(&mut rng)
+        .contains(&honest_peer));
+    assert!(builder
+        .peer_list()
+        .dishonest_peers()
+        .contains(&dishonest_peer));
+}
+
+#[test]
+fn register_era_validator_weights() {
+    let mut rng = TestRng::new();
+    let block = Block::random(&mut rng);
+    let mut builder = BlockBuilder::new(
+        block.header().block_hash(),
+        false,
+        1,
+        TimeDiff::from_seconds(1),
+        TimeDiff::from_seconds(1),
+    );
+
+    let mut validator_matrix = ValidatorMatrix::new_with_validator(ALICE_SECRET_KEY.clone());
+
+    builder.register_era_validator_weights(&validator_matrix);
```

Comment:
Can you add comments for each scenario tested here? Or split this test in 3 and have a relevant name for the test scenario?

Reply:
Done.